app/geo/gui/src/ts/twxplore/gui/geo/components/map/Map.tsx

```diff
@@ -97,68 +120,138 @@ const MapImpl: React.FunctionComponent = () => {
+      */
+      case MapFeatureState.RENDERED: {
+        for (const featureType of Object.values(FeatureType)) {
+          if (state.featuresByType[featureType].needsFilters) {
+            for (var x = 0; x < 3; ++x) {
+              dispatch(
+                addFilter(FeatureType[featureType as keyof typeof FeatureType])
+              );
+            }
+          }
+        }
         break;
       }
 
       case MapFeatureState.CLICKED: {
         /*
-      Features in this state have been clicked on the map. They now need to
-      LOAD in their child features.
+      Features in this state have been clicked on the map. A query for calling features within this feature is called with a limit.
+      This feature may contain even more features than the limit and thus is put into the 'CLICKED AND LOADING' state.
       */
+        totalDataCounter = 0;
+        latestDataCounter = 0;
 
         const clickedUris: string[] = [];
         for (const clickedFeature of featuresInState) {
           if (clickedFeature.type !== FeatureType.Transmission) {
             getFeaturesWithin({
               variables: {
-                query: {withinFeatureUri: clickedFeature.uri},
+                query: {
+                  withinFeatureUri: clickedFeature.uri,
+                },
+                limit: limit,
+                offset: 0,
               },
             });
           }
+
           clickedUris.push(clickedFeature.uri);
         }
+        dispatch(
+          changeMapFeatureState(
+            clickedUris,
+            MapFeatureState.CLICKED_AND_LOADING
+          )
+        );
+        break;
+      }
+
+      /*
+      Features in this state have been clicked and querying to find features within them has already been completed once.
+      The loading and finishedQuery variables are used to check if a query is still in progress. Once the query is done,
+      we compare the number of features gotten from the query (latestDataCounter) to the limit we gave it. If they are equal,
+      then more querying needs to be done as we need to query until we are not capped by the limit.
+      The totalDataCounter tracks how many features have been received and is used to offset for the next query.
+
+      */
+      case MapFeatureState.CLICKED_AND_LOADING: {
+        const clickedUris: string[] = [];
+        if (!loading && !finishedQuery) {
+          console.log("should be loading");
+          break;
+        }
+        if (loading) {
+          console.log("is loading");
+          break;
+        }
+        for (const clickedFeature of featuresInState) {
+          if (latestDataCounter === limit) {
+            console.log("Made it here");
+            getFeaturesWithin({
+              variables: {
+                query: {
+                  withinFeatureUri: clickedFeature.uri,
+                },
+                limit: limit,
+                offset: totalDataCounter,
```

Comment:
The offset is really a per-feature state. You should track this in the Redux store. You can either (1) subclass (sub-interface) the Feature interface from the API, add a member loadingOffset?: number, and store those extended things in the Redux state or (2) keep a separate (sparse) map of feature URI -> loading state in Redux, which only has features that are currently being expanded "within". The latter is probably simpler.